tc/core/polyhedral/schedule_tree_elem.h

```diff
@@ -31,16 +31,24 @@ namespace polyhedral {
 namespace detail {
 
 struct ScheduleTreeContext : public ScheduleTree {
+ public:
   static constexpr detail::ScheduleTreeType NodeType =
       detail::ScheduleTreeType::Context;
 
+ private:
   ScheduleTreeContext() = delete;
-  ScheduleTreeContext(const ScheduleTreeContext& eb)
-      : ScheduleTree(eb), context_(eb.context_) {}
   explicit ScheduleTreeContext(isl::set s)
       : ScheduleTree(s.get_ctx(), {}, NodeType), context_(s) {}
+
+ public:
+  ScheduleTreeContext(const ScheduleTreeContext& eb)
+      : ScheduleTree(eb), context_(eb.context_) {}
```

Comment:
If an outside user is not supposed to create objects directly,
then why is the copy constructor still exposed?